merlin/models/tf/metrics/evaluation.py

```diff
@@ -0,0 +1,292 @@
+#
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Adapted from source code: https://github.com/karlhigley/ranking-metrics-torch
+from typing import Optional, Sequence, Union
+
+import tensorflow as tf
+from keras.utils.tf_utils import is_tensor_or_variable
+from tensorflow.keras import backend
+from tensorflow.keras.metrics import Mean
+from tensorflow.keras.metrics import get as get_metric
+
+from merlin.models.tf.metrics import metrics_registry
+from merlin.models.tf.utils.tf_utils import get_candidate_probs
+
+MIN_FLOAT = 1e-16
+METRIC_PARAMETERS_DOCSTRING = """
+    predicted_candidates_probs: tf.Tensor
+        A tensor with shape (batch_size, n_items) corresponding to
+        the sorted predicted candidate probabilities. Where the item probability is
+        derived from the frequency of the items in the training set as follow:
+        `item_prob = item_freq_count / sum(item_freq_count)`
+    k : int
+        The cut-off for popularity metrics
+"""
+
+
+def novelty_at(
+    predicted_candidates_probs: tf.Tensor,
+    k: int = 5,
+) -> tf.Tensor:
+    """
+    Computes novely@K metric
+    ----------
+    {METRIC_PARAMETERS_DOCSTRING}
+    """
+    return -tf.math.log(predicted_candidates_probs[:, :k] + MIN_FLOAT)
```

Comment:
I think we should rename `MIN_FLOAT` to `EPSILON`, as there is another `MIN_FLOAT` constant in `merlin.models.utils.constants` used to downscore the false negatives